File: LongestCommonPrefix.java
public class LongestCommonPrefix {
    public static void main(String[] args) {
        String[] st={"flower", "flow", "flight"};
        System.out.println(longestPrefix(st));
    }
    public static String longestPrefix(String[] str){
        if(str.length == 0) return "";

        String prefix=str[0];
        for(int i=1;i<str.length;i++){
            while(str[i].indexOf(prefix) !=0){
                prefix = prefix.substring(0, prefix.length()-1);
            }
        }
        return prefix;
    }
}
//prefix=fl
//["flower", "flow", "flight"]

File: longest_common_prefix.py
def longestCommonPrefix(strs:list[str])->str:
    if len(strs)==0:
        return ""
    
    base=strs[0]
    
    for i in range(len(strs)):
        for word in strs[1:]:
            if i==len(word) or word[i] != base[i]:
                return base[0:i]
    return base

strs=["flowers","flow","flight"]
ans=longestCommonPrefix(strs)
print (ans)